src/application/common/logger.ts

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable, unmanaged } from 'inversify';
+import { LoggerWatcher } from './logger-watcher';
+
+export namespace LogLevel {
+    export const FATAL = 60;
+    export const ERROR = 50;
+    export const WARN = 40;
+    export const INFO = 30;
+    export const DEBUG = 20;
+    export const TRACE = 10;
+}
+
+export const ILoggerServer = Symbol("ILoggerServer");
+
+export interface ILoggerServer {
+    setClient(client: ILoggerClient | undefined): void;
+    setLogLevel(id: number, logLevel: number): Promise<void>;
+    getLogLevel(id: number): Promise<number>;
+    log(id: number, logLevel: number, message: string, params: any[]): Promise<void>;
+    child(obj: Object): Promise<number>;
+}
+
+export const ILoggerClient = Symbol("ILoggerClient");
+
+export interface ILogLevelChangedEvent {
+    oldLogLevel: number;
+    newLogLevel: number;
+}
+
+export interface ILoggerClient {
+    onLogLevelChanged(event: ILogLevelChangedEvent): void;
+}
+
+export const ILogger = Symbol("ILogger");
+
+export interface ILogger {
+    child(obj: Object): ILogger;
+    setLogLevel(logLevel: number): Promise<void>
+    getLogLevel(): Promise<number>;
+    trace(message: string, ...params: any[]): void;
+    debug(message: string, ...params: any[]): void;
+    info(message: string, ...params: any[]): void;
+    warn(message: string, ...params: any[]): void;
+    error(message: string, ...params: any[]): void;
+    fatal(message: string, ...params: any[]): void;
+}
+
+@injectable()
+export class Logger implements ILogger {
+
+    /* Log level for the loggers.  */
+    private _logLevel: Promise<number>[] = [];
+
+    /* Root logger has id 1.  */
+    private rootLoggerId = 1;
+
+    /* Default id is the root logger id.  */
+    private id: Promise<number> = Promise.resolve(this.rootLoggerId);
+
+    /**
+     * Build a new Logger.
+     *
+     * @param childOptions - The options to build the child with, see the
+     * bunyan documentation for more information.
+     */
+    constructor(
+        @inject(ILoggerServer) protected server: ILoggerServer,
+        @inject(LoggerWatcher) protected loggerWatcher: LoggerWatcher,
+        @unmanaged() childOptions?: Object) {
+
+        /* Creating a child logger.  */
+        if (childOptions !== undefined) {
+            this.id = server.child(childOptions);
+        }
+
+        /* Fetch the log level so it's cached in the frontend.  */
+        this.id.then(id => this._logLevel[id] = server.getLogLevel(id));
+
+        /* Update the root logger log level if it changes in the backend. */
+        loggerWatcher.onLogLevelChanged(event => {
+            this._logLevel[this.rootLoggerId] = Promise.resolve(event.newLogLevel);
```

Comment:
No child loggers are instance based, so if you create a child Logger in the frontend, it's frontend controlled only, same with backend.  
So you can't access a frontend created child logger from the backend and change it's level...